yt/frontends/art/data_structures.py

```diff
@@ -620,30 +616,24 @@ def create_field_info(self):
         pass
 
     @classmethod
-    def _is_valid(self, *args, **kwargs):
+    def _is_valid(cls, filename, *args, **kwargs):
         """
         Defined for the NMSU file naming scheme.
         This could differ for other formats.
         """
-        f = ("%s" % args[0])
         prefix, suffix = filename_pattern['particle_data']
-        if not os.path.isfile(f):
-            return False
-        if not f.endswith(suffix):
-            return False
-        if "s0" not in f:
+        if not filename.endswith(suffix) or "s0" not in filename:
             # ATOMIC.DAT, for instance, passes the other tests, but then dies
             # during _find_files because it can't be split.
             return False
-        with open(f, 'rb') as fh:
-            try:
-                amr_prefix, amr_suffix = filename_pattern['amr']
-                possibles = glob.glob(os.path.dirname(os.path.abspath(f))+"/*")
-                for possible in possibles:
-                    if possible.endswith(amr_suffix):
-                        if os.path.basename(possible).startswith(amr_prefix):
-                            return False
-            except Exception: pass
+
+        with open(filename, 'rb') as fh:
+            amr_prefix, amr_suffix = filename_pattern['amr']
+            for c in glob.glob(os.path.dirname(os.path.abspath(filename))+"/*"):
+                cname = os.path.basename(c)
+                if cname.endswith(amr_suffix) and cname.startswith(amr_prefix):
+                    return False
+            print("boh")
             try:
```

Comment:
this try block seems much weaker after deeper inspection : it's very easy to pass this without raising any exception, using fake data.